Automattic/wp-calypso
# Handle Markdown settings

#### Changes proposed in this Pull Request

* Adds a new toggle to Settings > Writing > Composing for enabling/disabling Markdown on posts/pages.
* Adds a new toggle to Settings > Discussion > Comments for enabling/disabling Markdown on comments.

<img width="400" alt="Screen Shot 2021-06-14 at 11 52 10" src="https://user-images.githubusercontent.com/1233880/121873858-183c4400-cd07-11eb-8ffd-b421cd9bc752.png">

<img width="400" alt="Screen Shot 2021-06-14 at 12 17 50" src="https://user-images.githubusercontent.com/1233880/121877285-c1386e00-cd0a-11eb-97cb-867d7c8bc757.png">

This helps with the effort of removing the "Jetpack > Settings" submenu needed for #51342, since right now the Markdown module for Atomic sites can be enabled only from that submenu.

#### Testing instructions
- Markdown for posts/pages - Simple sites:
  - Pick a simple site and go to `<SITE_DOMAIN>/wp-admin/options-writing.php`.
  - Disable "Use Markdown for posts and pages" if enabled.
  - Go to `/settings/writing` and pick the same simple site as above.
  - Make sure that "Write posts or pages in plain-text Markdown syntax" is disabled.
  - Enable it and save.
  - Go to `<SITE_DOMAIN>/wp-admin/options-writing.php`.
  - Make sure that "Use Markdown for posts and pages" is enabled.
- Markdown for posts/pages - Atomic sites:
  - Pick an Atomic site and go to Jetpack > Settings > Writing.
  - Disable "Write posts or pages in plain-text Markdown syntax" if enabled.
  - Go to `/settings/writing` and pick the same Atomic site as above.
  - Make sure that "Write posts or pages in plain-text Markdown syntax" is disabled.
  - Enable it and save.
  - Go to Jetpack > Settings > Writing.
  - Make sure that "Write posts or pages in plain-text Markdown syntax" is enabled.
- Markdown for comments - Simple sites:
  - Pick a simple site and go to `<SITE_DOMAIN>/wp-admin/options-discussion.php`.
  - Disable "Use Markdown for comments" if enabled.
  - Go to `/settings/discussion` and pick the same simple site as above.
  - Make sure that "Use Markdown for comments" is disabled.
  - Enable it (it should be auto-saved).
  - Go to `<SITE_DOMAIN>/wp-admin/options-discussion.php`.
  - Make sure that "Use Markdown for comments" is enabled.
- Markdown for comments - Atomic sites:
  - Pick an Atomic site and go to Jetpack > Settings > Discussion.
  - Disable "Enable Markdown use for comments" if enabled.
  - Go to `/settings/discussion` and pick the same Atomic site as above.
  - Make sure that "Use Markdown for comment" is disabled.
  - Enable it and save.
  - Go to Jetpack > Settings > Discussion.
  - Make sure that "Enable Markdown use for comments" is enabled.
